research/object_detection/meta_architectures/context_rcnn_meta_arch.py

```diff
@@ -268,7 +269,15 @@ def __init__(self,
         context_rcnn_lib.compute_box_context_attention,
         bottleneck_dimension=attention_bottleneck_dimension,
         attention_temperature=attention_temperature,
-        is_training=is_training)
+        is_training=is_training,
+        freeze_batchnorm=freeze_batchnorm)
+    
+    self._atten_projs = {'key': context_rcnn_lib.ContextProjection(
```

Comment:
We can encapsulate the whole context_feature_extract_fn functionality into a Keras Layer subclass. That way we don't have to manage these in meta_arch.